File: Data Structures & Algorithms/anagram-groups/submission-9.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        obj = defaultdict(list)
        res = []

        print(ord("a")-97)
        for item in strs:
            count = [0] * 26
            for char in item:
                count[ord(char)-97] += 1
            obj[tuple(count)].append(item)

        return list(obj.values())

File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        std::unordered_map<int,int> my_map;

        for (int i=0; i<nums.size(); i++) {
            my_map[nums[i]]++;
        }

        std::vector<std::vector<int>> bucket(nums.size()+1);

        for (auto& item : my_map) {
            bucket[item.second].push_back(item.first);
        }

        int count = 0;
        std::vector<int> res;

        for (int i = bucket.size()-1; i >= 0; i--) {
            for (int j=0; j < bucket[i].size(); j++) {
                if (count < k) {
                    res.push_back(bucket[i][j]);
                    count++;
                } else {
                    break;
                }
            }
        }

        return res;
    }
};
